Documentation/ApiOverview/CachingFramework/Index.rst

```diff
@@ -4,25 +4,155 @@
+to render non-cacheable content.
+
+When visiting a TYPO3 web site, TYPO3 knows the following states:
+
+- first time hit, page has never been rendered ("cache miss")
+- consecutive hit, page has been rendered before ("cache hit")
+
+In the first case, TYPO3 renders the complete page and writes cache entries as
+configured. In the second case, TYPO3 fetches those entries from the cache and
+delivers them to the user without re-triggering the rendering.
+
+In that second case, either the page is fully cached and directly delivered from
+the cache, or the page has non-cacheable elements on it. If a page has non-cacheable
+elements, TYPO3 first fetches the cached part of the page and then renders all
+dynamic parts.
+
+.. note::
+
+   The cache contains placeholders for dynamic elements to tell TYPO3 where to render
+   dynamic parts.
+
+
+.. hint::
+
+   For developers: If you are developing a plugin think about your plugins' cache lifetime.
+   Ideally, it can be fully cached, but if not, read the section about the caching framework
+   to learn how to leverage TYPO3's caching mechanism to cache your plugin for however long
+   you can - even 30seconds might improve performance in some scenarios.
+
+
+Caching Variants - or: What is a "cache hash"?
+----------------------------------------------
+
+TYPO3 ideally delivers fully cached pages for maximum performance. However, in scenarios where
+the same page will deliver different content depending on URL parameters, TYPO3 needs a possibility
+to identify these "variants" and cache each of them differently. For example, if you have a
+news plugin and a detail page, the detail page is different for every news entry.
+
+To identify the variant, TYPO3 combines a set of parameters and generates a hash value as identifier.
+These parameters include by default:
+
+- `id`: The current page ID
+- `type`: The current page type (`typeNum`)
+- `groupIds`: The user groups of the logged in user (if no user is logged in: 0, -1 as default values)
+- `MP`: The mount point identifier
+- `site`: The current site and base URL
+- `staticRouteArguments`: Any route argument configured in the routing configuration and resolved in the current request
+- `dynamicArguments`: Any :php:`$_GET` parameters influencing the rendered page
+
+Imagine the following URL `https://example.com/news/?tx_example_news[id]=123` displaying the news with ID `123`.
+If TYPO3 would cache that page with that parameter without any security mechanisms, it would open a potential
+denial of service attack as an unlimited amount of cache entries could be generated by adding arbitrary parameters.
+To avoid that, TYPO3 generates a so-called `cHash` parameter, which is a hash that basically signs the valid parameters
+for that request. So any parameter that validly influences the rendered page needs to be part of that `cHash`.
+
+With :ref:`Routing` you can configure TYPO3 not to display the `cHash` in your URLs in most cases.
```

Comment:
Here we should add a little explanation why:

> Routing adds an explicit mapping of incoming readable URL slugs to internal parameter values. This both adds an additional layer for validating slugs as well as reduce the parameters to a limited set of values.

(Or something similar. ;-) )

Reply:
👍